src/client.js

```diff
@@ -7,35 +7,28 @@ import productConnectionQuery from './product-connection-query';
 import collectionQuery from './collection-query';
 import collectionConnectionQuery from './collection-connection-query';
 import ProductHelpers from './product-helpers';
-
-function fetchAllPages(paginatedModels, client) {
-  return client.fetchNextPage(paginatedModels).then(({model}) => {
-    paginatedModels.push(...model);
-
-    if (!model[model.length - 1].hasNextPage.valueOf()) {
-      return paginatedModels;
-    }
-
-    return fetchAllPages(paginatedModels, client);
-  });
-}
+import checkoutQuery from './checkout-query';
+import fetchAllPages from './fetch-all-pages';
```

Comment:
Shouldn't this rely on the feature you pushed up stream into the graphql client?